src/libostree/ostree-repo-pull.c

```diff
@@ -1571,20 +1571,19 @@ _ostree_repo_sign_verify (OstreeRepo *repo,
 {
   /* list all signature types in detached metadata and check if signed by any? */
   g_auto (GStrv) names = ostree_sign_list_names();
-  g_autoptr (GError) verification_error = NULL;
-
-  glnx_throw (&verification_error, "signed with unknown key");
+  guint n_invalid_signatures = 0;
+  g_autoptr (GError) last_sig_error = NULL;
+  gboolean found_sig = FALSE;
 
   for (char **iter=names; iter && *iter; iter++)
     {
       g_autoptr (OstreeSign) sign = NULL;
       g_autoptr (GVariant) signatures = NULL;
       const gchar *signature_key = NULL;
       GVariantType *signature_format = NULL;
-      g_autoptr (GError) local_error = NULL;
 
-      if ((sign = ostree_sign_get_by_name (*iter, &local_error)) == NULL)
-        continue;
+      if ((sign = ostree_sign_get_by_name (*iter, error)) == NULL)
+        return FALSE;
```

Comment:
Oh wow yes, the test suite relies on having each one try to parse the key type and ignore failures.

How about we change things so that instead it's e.g.:

```
verification-key=ed25519:<blah>
```

Reply:
hm... good idea for adding prefix for `verification-key`, I think that `verification-file` should to use it as well. But that is not related to the current case -- for instance `ed25519` may work without those settings if public signatures are placed inside the "well-known" directories.

Let me to describe why I using the current approach:
I expect that `commitmeta` may contain several signature types with several signatures for each type. Currently we query the sign module for which signature types we are supporting and try to check if there are any of them in detached metadata -- one by one. I didn't wanted to propose separate API call for checking the engine by signature name from metadata.

So, for example in case if the commit has signed with only valid `ed25519` -- we should be able to check it. But with proposed changes it is leading that only `dummy` engine is queried and stop the check for the rest of engines since the detached `commitmeta` file doesn't contains signature of `dummy` type.
As a result it leads to  the failure of `tests/test-signed-pull.sh` for all `ed25519` test cases.

The similar problem we have if there is a `dummy` signature configuration is failed for some reason while loading public keys -- there will be no chance for other signature types to check if the commit is signed with correct signature.

The current behavior is designed to be able to work with multiple signature types -- and allows to sign the commit with several signatures and several signature types in the same time. I expect it will be useful for switching between signature types as soon as we will have some additional engines besides the `dummy` and `ed25519`. To be honest the `dummy` module has been added mostly for prototyping and checking the multiple signature types work.